README.md

```diff
@@ -40,9 +41,153 @@ Spline aims to fill a big gap within the Apache Hadoop ecosystem. Spark jobs sho
+-   [```za.co.absa.spline:client-web:0.4.0```](https://repo1.maven.org/maven2/za/co/absa/spline/client-web/0.4.0/) (optional)
+
+(REST Server and Web Client modules are also available as [Docker containers](https://hub.docker.com/u/absaoss))
+
+-or-
+
+#### Build Spline from the source code
+1.  Make sure you have JDK 8, Maven and NodeJS installed.
+
+1.  Get and unzip the Spline source code:
+    ```shell script
+    wget https://github.com/AbsaOSS/spline/archive/release/0.4.0.zip
+    unzip 0.4.0.zip
+    ```
+1.  Change the directory:
+    ```shell script
+    cd spline-release-0.4.0
+    ```
+1.  Run the Maven build:
+    ```shell script
+    mvn install -DskipTests
+    ```
+
+## Install ArangoDB
+Spline server requires ArangoDB to run.
+Please install _ArangoDB 3.5+_ according to the instructions [here](https://www.arangodb.com/docs/stable/getting-started-installation.html)
+
+If you prefer a Docker image there is a [Docker repo](https://hub.docker.com/_/arangodb/) as well.
+```shell script
+docker pull arangodb:3.5.1
+```
+
+## Create Spline Database
+```shell script
+java -jar admin/target/admin-0.4.0.jar db-init arangodb://localhost/spline
+```
+
+## Start Spline Server
+The easiest way to spin up the Spline server is to use Docker:
+
+```shell script
+docker container run \
+      -e spline.database.connectionUrl=arangodb://172.17.0.1/spline \
+      -p 8080:8080 \
+      absaoss/spline-rest-server
+```
+
+Or you can deploy it as a WAR-file into any Java compatible Web-Container, e.g. Tomcat.
+You can find a WAR-file in the Maven repo here:
+[```za.co.absa.spline:rest-gateway:0.4.0```](https://repo1.maven.org/maven2/za/co/absa/spline/rest-gateway/0.4.0/)
+
+The server exposes the following REST API:
+-   Producer API (`/producer/*`) 
+-   Consumer API (`/consumer/*`)
+
+... and other useful URLs:
+-   Running server version information: [/about/version](http://localhost:8080/about/version)
+-   Producer API Swagger documentation: [/docs/producer.html](http://localhost:8080/docs/producer.html) 
+-   Consumer API Swagger documentation: [/docs/consumer.html](http://localhost:8080/docs/consumer.html) 
+
+
+## Start Spline UI
+
+Spline web client can be started using 3 diffrent ways:
+
+- Docker: 
+
+```shell script
+docker container run \
+      -e -e spline.consumer.url=http://172.17.0.1:8080/consumer \
+      -p 9090:8080 \
+      absaoss//spline-web-client
+```
+
+- Java compatible Web-Container:
+
+You can find the WAR-file of the Web Client in the repo here:
+[```za.co.absa.spline:client-web:0.4.0```](https://repo1.maven.org/maven2/za/co/absa/spline/client-web/0.4.0/)
+
+- Node JS application : 
+
+Download [```node.js```](https://nodejs.org/en/) then install [```@angular/cli```](https://www.npmjs.com/package/@angular/cli) to run `ng serve` or `ng-build` command.
+
+You can find the documentation of this module in [ClientUI](client-ui/README.md).
+
+## Check the result in the browser
+http://localhost:9090
+
+
```

Comment:
Suggestion:

Now you can try to run some examples --link to examples--

Reply:
added the link to the examples